### PR TITLE
Update jar, class name

### PR DESCRIPTION
**Pull request by evacchi**
Remove the original jar use current `latest`: https://github.com/kiegroup/drools-ansible-rulebook-integration/commit/490aae9f10fce891f74b4b119d37c157ec78fb7c

We have set-up GitHub actions to publish jars at known locations.

- `latest` tag will always point to the latest successful build 
   curl -LO https://github.com/kiegroup/drools-ansible-rulebook-integration/releases/download/latest/drools-ansible-rulebook-integration-runtime-1.0.0-SNAPSHOT.jar

- `$version` will always point to the given tag; e.g.:
curl -LO https://github.com/kiegroup/drools-ansible-rulebook-integration/releases/download/$version/drools-ansible-rulebook-integration-runtime-1.0.0-SNAPSHOT.jar (currently no tagged release)

With this PR I am pushing the jar from `latest` https://github.com/kiegroup/drools-ansible-rulebook-integration/commit/490aae9f10fce891f74b4b119d37c157ec78fb7c but now that URLs are stable a script could be implement to download the jar.

The name is currently fixed to `drools-ansible-rulebook-integration-runtime-1.0.0-SNAPSHOT.jar` regardless of the version, but we can change that to any desired name.

We have also changed the packaging strategy and now the jar is down to a nice **6megs** instead of those previous **24megs**. This needs some testing to ensure that it does work correctly as before.

### EDIT: 

Good News: The jar works!
(slightly) Bad News: So this PR shows evidence of some issues that we hadn't verified yet :-)

In particular, when a fact is not matched in any rule, then it is immediately discarded, therefore `get_facts()` won't return it.

We need to discuss this because in this case we are behaving as expected: shall we just update the tests?

### CODE REVIEW

**Review by mariofusco — APPROVED** [review]

**Comment by mkanoor** [comment]
@evacchi Can you please look at the test failures, I think the test_retract fact is failing along with others.

**Comment by evacchi** [comment]
yes @mkanoor this is a consequence of how we are now handling facts, because we decided to discard those facts that are not matched by any rule now those tests are failing. This is intended behavior; however, if we believe that they should be returned, we should have a chat about it! @mariofusco

**Comment by mkanoor** [comment]
@mariofusco We had implemented host based facts which depend on facts be stored for later comparison. Those facts that are initially posted will never match anything so if we delete facts because they don't match anything we won't be able to use the host based comparison. I know durable rules doesn't cache facts and events which don't match anything we were tapping the feature in Drools to keep the facts around but not the events when things don't match.

**Comment by evacchi** [comment]
so if I understand correctly, we should purge only _events_ not facts, right?

**Comment by mariofusco** [comment]
> @mariofusco We had implemented host based facts which depend on facts be stored for later comparison. Those facts that are initially posted will never match anything so if we delete facts because they don't match anything we won't be able to use the host based comparison. 

@mkanoor I believe that this statement is incorrect or at least there is a misunderstanding that is necessary to clarify.

What Drools does is removing facts and events (at the moment we didn't implement any difference on this regard) that not only don't match any rule at insertion time but that cannot match any rule at any time in future. Let me make a practical example: if in a rule set there is only a rule with the following condition:

```
condition:
  all:
    - myFact << fact.host == "H1"
```
and you insert a fact with host attribute equal to "H2", then we know that this fact not only is not matching the rule now, but it can never match it in future, so there is no reason to keep it and we can safely throw it anyway. Conversely if you insert the same fact with host attribute equal to "H2" into a an executor running a rule set with the following rule:
  
```
condition:
  all:
    - myFact << fact.host IsDefined
    - myEvent << event.host == fact.myFact.host 
```
then the fact cannot and won't be automatically evicted by Drools because, even if at the moment there is no event coming from the same host "H2" of the inserted fact, we cannot exclude that this could happen in future thus finally generating a match.

In essence we automatically evict only facts and events for which there is at least one constraint on a fixed value (what is called an `alpha constraint` in Drools) that cannot be satisfied while, if that fact is part of a join condition (a `beta constraint`), it won't be removed.

To recap if you're concerned that the automatic eviction could throw away facts that could be potentially necessary to produce a future match I can ensure that this won't happen. Conversely if you also want to keep those facts regardless if they could produce a match or not, maybe because you also want to use Drools working memory as a queryable cache (?), then we will need to change a bit Drools logic to enforce the retention of those facts.

Please let us know how you want to proceed on this.

**Comment by mkanoor** [comment]
@evacchi We would have to fix the tests/test_ruleset.py file where we are are having test failures.

**Comment by evacchi** [comment]
thanks I have pushed the changes you sent to me @mkanoor
